### ISSUE TITLE
Improve repairing strings missing quotes like `0.0.1` and `ES2020`

### ISSUE THREAD

**Issue by liuyib** [josdejong/jsonrepair#120]
Hi @josdejong, thanks a lot for such a great library, really helps me out.

I have found that jsonrepair will fail in some cases, I wonder if this is expected?

1. A non-number appears after a number
![image](https://github.com/josdejong/jsonrepair/assets/38221479/f703500c-5685-4528-b5e8-143d8f36f239)
![image](https://github.com/josdejong/jsonrepair/assets/38221479/0aa503eb-67e1-4215-b643-42fd57feddd8)
2. Any character that appears before/after boolean
![image](https://github.com/josdejong/jsonrepair/assets/38221479/f5cf27c4-0f69-4751-bcbf-71d68a98e8ce)
![image](https://github.com/josdejong/jsonrepair/assets/38221479/1359a7aa-7bbb-416b-8a65-5d16a89f0d1c)
3. `-+`
![image](https://github.com/josdejong/jsonrepair/assets/38221479/e004a871-b518-4c44-a335-53b7e5fcd50c)
![image](https://github.com/josdejong/jsonrepair/assets/38221479/9269b7c4-b508-48a6-a70e-6546d696488d)

I want to open a PR to fix these, can you confirm which ones need to be fixed? thx @josdejong

**Comment by josdejong** [comment]
@liuyib thanks for your suggestions. Are these cases that you really came across in the real-world, or are they made up? How would you expect these examples to look like after being repaired?

**Comment by liuyib** [comment]
`1` is came across the real-world:

tsconfig.json:
![image](https://github.com/josdejong/jsonrepair/assets/38221479/c995a6b2-cb1a-4f86-8539-03a297f5964a)

package.json:
![image](https://github.com/josdejong/jsonrepair/assets/38221479/3510dd01-1486-4da1-a034-d0dd604fa9a0)


`2, 3` are made up.

**Comment by josdejong** [comment]
Ahh, those two examples make a lot of sense indeed. Let's see if we can improve `jsonrepair` to handle those cases!

Do you want to work this out in a PR? If so, it's important to know that the library contains two implementations: the regular under `src/regular`, and a streaming implementation under `src/streaming`. Both share the same set of tests. Easiest is to first implement a solution in the regular version, and when that works port it to the streaming version.

**Comment by liuyib** [comment]
Yes, I'd like to handle those cases. I will try to fix them, thanks for your development guidance :)

**Comment by josdejong** [comment]
👍 thanks!

**Comment by hey68you** [comment]
Hi @josdejong @liuyib - I'm facing a similar issue - is there a plan to fix this?
![image](https://github.com/josdejong/jsonrepair/assets/31699161/6b829a81-9110-4056-9dbd-5387e35d1b3b)
![image](https://github.com/josdejong/jsonrepair/assets/31699161/c5d3ab4a-074f-4605-bac1-597c3fa9ea47)

Repair doesn't seem to work for uuid type strings or file name with dash chars or is the problem the .wav file type ?
unless I manually added the the qoutes
`{memo_id:746de9ad-d4ff-4c66-97d7-00a92ad46967, audio_file:19525_1_low.wav}`

vs.

`{memo_id:"746de9ad-d4ff-4c66-97d7-00a92ad46967", audio_file:"19525_1_low.wav"}`

**Comment by liuyib** [comment]
Sorry, I don't have time to trace this issue any more. Can anyone help? Thanks

**Comment by josdejong** [comment]
Thanks for the update @liuyib, no problem.

**Comment by josdejong** [comment]
> Hi @josdejong @liuyib - I'm facing a similar issue - is there a plan to fix this?

The plan is indeed to fix this. Help would be very welcome.

In short: we want to fix the following three cases of values that should be turned into strings:

```
{
  target: ES2020,  
  version: 0.0.1,
  uuid: 746de9ad-d4ff-4c66-97d7-00a92ad46967
}
```

**Comment by hey68you** [comment]
Great - looking forward - I don't think I can help at this time - would be great if possible to deal with the case where the property  value has spaces: 

`{
    usercomments: hello world,
    ...
}`

**Comment by josdejong** [comment]
👍 

> would be great if possible to deal with the case where the property value has spaces

That works already, have you tried it out?

**Comment by josdejong** [comment]
Fixed now in `v3.7.0`
